Gordon360/Services/NewsService.cs

```diff
@@ -153,9 +173,156 @@ public StudentNewsViewModel EditPosting(int newsID, StudentNews newData)
             newsItem.Subject = newData.Subject;
             newsItem.Body = newData.Body;
 
+            if (newData.Image != null)
+            {
+                UploadImage(newData.Image, newsID);
+            }
+
+            //If the image property is null, it means that either the user
+            //chose to remove the previous image or that there was no previous
+            //image (RemoveImage is designed to handle this).
+            else
+            {
+                RemoveImage(newsItem.SNID);
+                newsItem.Image = newData.Image;
+            }
+
             _unitOfWork.Save();
             
-            return (StudentNewsViewModel)newsItem;
+            return newsItem;
+        }
+
+        /// <summary>
+        /// Takes a filepath for an image, navigates to it, collects the raw data
+        /// of the file and converts it to base64 format. 
+        /// 
+        /// The base64 data will not include the first part of a base64 image ("data:image/...").
+        /// This is because this part is removed in every image before being submitted, and it
+        /// is readded in the frontend before being displayed.
+        /// 
+        /// This helper function does not perform any error checking; every place that calls it
+        /// checks that the path is not empty. Theoretically if it isn't empty it's certainly a valid path.
+        /// </summary>
+        /// <param name="imagePath">The path to the image</param>
+        /// <returns>The base64 content of the image</returns>
+        public string GetBase64ImageDataFromPath(string imagePath)
+        {
+            using (System.Drawing.Image image = System.Drawing.Image.FromFile(imagePath))
+            using (MemoryStream data = new MemoryStream())
+            {
+                image.Save(data, image.RawFormat);
+                byte[] imageBytes = data.ToArray();
+                string base64Data = System.Convert.ToBase64String(imageBytes);
+                return base64Data;
+            }
+        }
+
+        /// <summary>
+        /// Iterates through a collection of news items that have just been queried from
+        /// the database, replacing the Image entry of each (which contains the path to an image)
+        /// with the image itself. This is done because when the frontend makes requests for news items,
+        /// it needs the image data and not the path, but we store the path in the DB for efficiency and
+        /// storage reasons.
+        /// </summary>
+        /// <param name="items">The collection of student news entries</param>
+        /// <returns>The collection of entries with base64 data in their image variable</returns>
+        private IEnumerable<StudentNewsViewModel> InsertImageData(IEnumerable<StudentNewsViewModel> items)
```

Comment:
Another, much simpler, way that this could be made more idiomatic and reusable would be to let it map from an image path to image data, so like:
```suggestion
        public string RetrieveImageFromPath(string imagePath)
```

Then, you could call it very simply using the LINQ `Select()` method, e.g.:
```
IEnumerable<StudentNewsViewModel> items = RawSqlQuery<StudentNewsViewModel>
  .query("NEWS_NEW")
  .Select(n => n.Image = RetrieveImageFromPath(n.Image));
```

This way, you haven't hard-coded the method to only work for StudentNews, and only when it's inside an `IEnumerable`, meaning you could use this anywhere that someone needs to retrieve an image from the file server. It also changes the very non-C#-like `InsertImageData(items);` into the much more C#-like (and more similar to our codebase) `item.Image = RetrieveImageFromPath(imagePath);`. With the former, I have to look at the implementation of the method to realize that it's mutating the items I pass into, which is unexpected and kinda odd.